js-project-recipe-library by Oscar

osckli990 opened the pull request:
#https://js-project-recipe-library.netlify.app

My recipe library:
- basic design following the figma template
- filtering and sorting options with the ability to combine them
- pagination!

Still got issues to fix, which will be updated this or the coming week

osckli990 committed:
startup

osckli990 committed:
html setup

osckli990 committed:
styles and html structure

osckli990 committed:
added a bunch of stuff, tried testing with Jest, doesnt work, added javascript, styled some more, finished HTML. will add media query later

osckli990 committed:
added more testing which doesnt work. completed js, started adding media query

osckli990 committed:
tried changing to checkbox instead of radio, resulting in some unexpected behavior. The site is now responsive

osckli990 committed:
started prepping html and js for the weeks requirements

osckli990 committed:
removed code from previous week and added some recipes through JS

osckli990 committed:
added some more recipes and styling

osckli990 committed:
added the filter in js and cleaned up html from previous week with temp text box

osckli990 committed:
some trouble with js, rewriting

osckli990 committed:
fixed and redid alot of JS, which now works and includes sort by time and cost. redid cost to be fewer buttons, when selecting diets, multiple can be selected as well as reloading the original recipes if unchecked

osckli990 committed:
added some comments and included the netlify link in the README

osckli990 committed:
removed previous testing i couldn't get to work. added a randomized recipe for a button i created earlier. created a very specific if statement for a combination of sorting that would result in no recipes being shown. will change when i have time

osckli990 committed:
resolved a bug where are recipes would not be displayed if a diet was checked and unchecked. was a problem with sorting by time, as the value of the recipe was higher than the max sorting value of 60. added comment to HMTL

osckli990 committed:
succeded in writing a more general if statement for no valid recipes

osckli990 committed:
save before checkbox test

osckli990 committed:
changed css to mobile-first, improved JS.

osckli990 committed:
eeehhhh

osckli990 committed:
recipes now displaying and working, filters not working will re-work functions

osckli990 committed:
added ALOT of code, removing and fixing to try and get pagination to work, it currently kinda works with loading new recipes but contains duplicates of stored recipes. Reworked the update function to contain the selected- diet/cost/time to make the code shorter. currently iäm also re-declaring let storedRecipes a multitude of times. could be improved, probably. added links to innerHTML. will need to comment and re-comment all code to better understand it and make it easier for others

osckli990 committed:
fixed the issue of duplicates

osckli990 committed:
possible fix to limit api fetches

osckli990 committed:
not working?

osckli990 committed:
pagination now works

osckli990 committed:
commented out

osckli990 committed:
reworked spinner and loader, tweaked displayed price and ingredients, ...

osckli990 committed:
recipes no longer load when a filter is active, wit hthe exception of the default time filter

osckli990 committed:
for code review

osckli990 committed:
fixed fetch still fetching despite sorting enabled

osckli990 committed:
decided to change displayed ingredient amount to fractions instead of decimals, need to comment code, shall try writing code more concise and learn how that works. This is mostly a message to myself

osckli990 committed:
convert small weird decimal values to simple fraction. i.e 0.27 -> 1/3. leave larger values be. complexsearch seems to return recipes with a different structure to randomsearch

osckli990 committed:
removed local node causing file overload?

osckli990 committed:
changed decimalToFraction to use specific and smaller denominators. Re-worked updateRecipes and fetchData to be MUCH more concise utilizing .map, .filter, ternary operator and the spread operator (...). these are listed in the order from being able to understand to not understand

TiagoAbrahamsson commented on script.js:
```diff
@@ -219,7 +218,7 @@ function checkScroll() {
     checkMix.some(checkbox => checkbox.checked) || // Diet filters
     costMix.some(checkbox => checkbox.checked) || // Cost sorting
-    (selectedTime && selectedTime === "60") // Only block if time is selected & NOT "60-min"
+    (selectedTime && selectedTime === "60") // Only block if time is selected & not "60-min"
```
You have selectedTime written here twice which doesn't seem to be neccessary.

osckli990 replied:
Strange, in my code there's only one line. hmm, i'll commit again i guess

TiagoAbrahamsson reviewed (commented):
Your code is well structured and logically sound.

The website looks good and resembles the figma design well.

osckli990 committed:
duplicate line

HIPPIEKICK reviewed (changes requested):
Good job with this project Oscar!
You've chosen descriptive function names and I can see you spent time on following the design - kudos 👍

Some things to work more on:
Please be consistent with how you define your functions → stick to arrow functions as much as you can. Also, please remove console.log from production code.

Looking through your code, I can’t see you’ve met this requirement 👀:
- Your page should show a useful message to the user in case the daily API quota has been reached

Also, I found a little bug. I’d expect when I check vegetarian and gluten-free that I’d get recipes that are just that - vegetarian and gluten-free. Buuut, I get a recipe including chicken. Please fix this or make sure that only one diet filter is chosen at a time. For the stretch goal regarding this, it’s enough for me if the filter goes well together with the sortings. Speaking of stretch goals, I can “only” see that you’ve done 1-3 which is totally fine but I just wanted to make sure I’m not missing something if you’re aiming for VG. Let me know!

osckli990 committed:
changed diets to be AND instead of OR

osckli990 committed:
fixed all recipes loading at once?

osckli990 commented:
Thank you!

Right, arrow functions. I don't know why i didn't use it consistently.

- I did have a rudimentary limit, based on calls, changed it to when the API throws the limit error. It should have displayed a message, but now it definetly does

- Fixed bug with multiple selected diets. It was set to AND and is now OR. So "vegetarian" won't show meat recipes

- The infinite-scrolling, the 4th requirement, has been fixed and improved. The recipes still loaded before, but it took a long while. improved "loading" icon and changed when it should fetch. a thought being if a complexSearch takes longer time to fetch. Had an error that i've failed to reproduce, where all recipes where fetched at once on scroll-down. Should be fixed

HIPPIEKICK commented:
> It was set to AND and is now OR. So "vegetarian" won't show meat recipes

You should probably fix the styling as well now then so that it's clear that the user picks one diet (if you changed the logic to or instead of and 😇).

Apart from that, it looks super!

osckli990 commented:
i think i meant it was OR, so if vegetarian and gluten-free was selected it would still show meat, since one of the diets are correct. it is now AND, so if the same diets are selected, it only shows recipes containing both diets
